.circleci/config.yml

```diff
@@ -0,0 +1,154 @@
+# Scala CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/sample-config/ for more details
+#
+# Use the chisel3-tools image to build and run FIRRTL and chisel tests.
+defaultImageUserWDEnv: &defaultImageUserWDEnv
+  docker:
+    - image: ucbbar/chisel3-tools
+      user: "chisel"
+
+  working_directory: ~/repo
+
+  environment:
+    # Customize the JVM maximum heap limit
+    JVM_OPTS: -Xmx3200m
+    TERM: dumb
+    CHISEL_REV: origin/master
+    FIRRTL_REPO: git@github.com:freechipsproject/firrtl.git
+    FIRRTL_BRANCH: master
+    FIRRTL_REV: master
+    SBT_ARGS: ""
+
+
+version: 2
+
+jobs:
+  build-prep:
+    <<: *defaultImageUserWDEnv
+
+    steps:
+
+    # Perform a "default" checkout to get ssh set up.
+      - checkout:
+          path: chisel3
+
+      - run:
+          command: |
+            date > date.prep
+            printenv
+            ls -l
+            git clone --depth 10 --branch $FIRRTL_BRANCH "$FIRRTL_REPO" firrtl && (cd firrtl && git checkout $FIRRTL_REV)
+            echo $FIRRTL_REV && (cd firrtl && git log -1 > ../firrtl.log)
+
+
+      - persist_to_workspace:
+          root: /home/chisel
+          paths:
+            - repo
+            - .ivy2
+            - .m2
+            - .sbt
+
+  build-firrtl:
+    <<: *defaultImageUserWDEnv
+
+    steps:
+      - attach_workspace:
+          at: /home/chisel
+
+      # publish FIRRTL
+      - run:
+          command: |
+            date > date.firrtl
+            (cd firrtl && cat /dev/null | sbt $SBT_ARGS +publishLocal)
```

Comment:
I believe that's from the CircleCI recommended sbt invocation. If sbt encounters errors trying to parse build.sbt, it tries to read from stdin and will hang. This ensures that if sbt tries to read from stdin, it will receive EOF and exit.

Reply:
Sounds good. Thanks for the clarification.